From f41fd8808accbe9c7771647d3beddc54fc5e7036 Mon Sep 17 00:00:00 2001
From: Steve Anton <anxuiz.nx@gmail.com>
Date: Mon, 29 Feb 2016 18:13:58 -0600
Subject: [PATCH] Add PlayerInitialSpawnEvent

For modifying a player's initial spawn location as they join the server

diff --git a/src/main/java/com/destroystokyo/paper/event/player/PlayerInitialSpawnEvent.java b/src/main/java/com/destroystokyo/paper/event/player/PlayerInitialSpawnEvent.java
new file mode 100644
index 0000000..d1d6f33
--- /dev/null
+++ b/src/main/java/com/destroystokyo/paper/event/player/PlayerInitialSpawnEvent.java
@@ -0,0 +1,43 @@
+package com.destroystokyo.paper.event.player;
+
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+
+public class PlayerInitialSpawnEvent extends PlayerEvent {
+    private static final HandlerList handlers = new HandlerList();
+    private Location spawnLocation;
+
+    public PlayerInitialSpawnEvent(final Player player, final Location spawnLocation) {
+        super(player);
+        this.spawnLocation = spawnLocation;
+    }
+
+    /**
+     * Gets the current spawn location
+     *
+     * @return Location current spawn location
+     */
+    public Location getSpawnLocation() {
+        return this.spawnLocation;
+    }
+
+    /**
+     * Sets the new spawn location
+     *
+     * @param spawnLocation new location for the spawn
+     */
+    public void setSpawnLocation(Location spawnLocation) {
+        this.spawnLocation = spawnLocation;
+    }
+
+    @Override
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}
-- 
2.10.0.windows.1

